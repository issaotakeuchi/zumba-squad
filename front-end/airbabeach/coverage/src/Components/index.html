
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/Components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/Components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">26.51% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>236/890</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>10/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">13.33% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/30</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">26.51% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>236/890</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Card.jsx"><a href="Card.jsx.html">Card.jsx</a></td>
	<td data-value="9.48" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 9%"></div><div class="cover-empty" style="width: 91%"></div></div>
	</td>
	<td data-value="9.48" class="pct low">9.48%</td>
	<td data-value="137" class="abs low">13/137</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="9.48" class="pct low">9.48%</td>
	<td data-value="137" class="abs low">13/137</td>
	</tr>

<tr>
	<td class="file high" data-value="Footer.jsx"><a href="Footer.jsx.html">Footer.jsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	</tr>

<tr>
	<td class="file medium" data-value="Form.jsx"><a href="Form.jsx.html">Form.jsx</a></td>
	<td data-value="53.08" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 53%"></div><div class="cover-empty" style="width: 47%"></div></div>
	</td>
	<td data-value="53.08" class="pct medium">53.08%</td>
	<td data-value="292" class="abs medium">155/292</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="18" class="abs low">8/18</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="16" class="abs low">2/16</td>
	<td data-value="53.08" class="pct medium">53.08%</td>
	<td data-value="292" class="abs medium">155/292</td>
	</tr>

<tr>
	<td class="file low" data-value="Header.jsx"><a href="Header.jsx.html">Header.jsx</a></td>
	<td data-value="8.65" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 8%"></div><div class="cover-empty" style="width: 92%"></div></div>
	</td>
	<td data-value="8.65" class="pct low">8.65%</td>
	<td data-value="104" class="abs low">9/104</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="8.65" class="pct low">8.65%</td>
	<td data-value="104" class="abs low">9/104</td>
	</tr>

<tr>
	<td class="file low" data-value="HeartIcon.jsx"><a href="HeartIcon.jsx.html">HeartIcon.jsx</a></td>
	<td data-value="8.88" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 8%"></div><div class="cover-empty" style="width: 92%"></div></div>
	</td>
	<td data-value="8.88" class="pct low">8.88%</td>
	<td data-value="90" class="abs low">8/90</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="8.88" class="pct low">8.88%</td>
	<td data-value="90" class="abs low">8/90</td>
	</tr>

<tr>
	<td class="file low" data-value="Map.jsx"><a href="Map.jsx.html">Map.jsx</a></td>
	<td data-value="24.32" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 24%"></div><div class="cover-empty" style="width: 76%"></div></div>
	</td>
	<td data-value="24.32" class="pct low">24.32%</td>
	<td data-value="37" class="abs low">9/37</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="24.32" class="pct low">24.32%</td>
	<td data-value="37" class="abs low">9/37</td>
	</tr>

<tr>
	<td class="file low" data-value="SearchBar.jsx"><a href="SearchBar.jsx.html">SearchBar.jsx</a></td>
	<td data-value="5.45" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.45" class="pct low">5.45%</td>
	<td data-value="165" class="abs low">9/165</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="5.45" class="pct low">5.45%</td>
	<td data-value="165" class="abs low">9/165</td>
	</tr>

<tr>
	<td class="file low" data-value="StarRate.jsx"><a href="StarRate.jsx.html">StarRate.jsx</a></td>
	<td data-value="21.95" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 21%"></div><div class="cover-empty" style="width: 79%"></div></div>
	</td>
	<td data-value="21.95" class="pct low">21.95%</td>
	<td data-value="41" class="abs low">9/41</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="21.95" class="pct low">21.95%</td>
	<td data-value="41" class="abs low">9/41</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-03-10T22:28:29.759Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    