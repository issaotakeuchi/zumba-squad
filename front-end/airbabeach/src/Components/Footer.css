.footer {
  width: 100%;
  height: 3.5rem;
  padding: 0 3.125rem;
  background-color: var(--green-light);
  position: fixed;
  bottom: 0px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  color: white;
  font-family: 'Roboto';
  font-size: .875rem;
}

.footer .footerParagraph {
  color: var(--gray-light);
}

.footer .socialMidia {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  gap: 1.875rem;
}

.footer .socialMidia a {
  color: var(--gray-light);
}

@media (max-width: 768px) {
  .footer {
    padding: 0 2rem;
  }
  .footer .socialMidia {
    gap: 1rem;
  }
}

@media (max-width: 414px) {
  .footer {
    -webkit-box-pack: start;
        -ms-flex-pack: start;
            justify-content: flex-start;
    padding: 0 1.2rem;
  }
  .footer .socialMidia {
    display: none;
  }
}
/*# sourceMappingURL=Footer.css.map */