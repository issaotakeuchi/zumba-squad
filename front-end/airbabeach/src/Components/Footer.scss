.footer {
    width: 100%;
    height: 3.5rem;
    padding: 0 3.125rem;
    background-color: var(--green-light);
    position: fixed;
    bottom: 0px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-family: 'Roboto';
    font-size: .875rem;

    .footerParagraph {
        color: var(--grey-light);
    }

    .socialMidia {
        display: flex;
        gap: 1.875rem;

        a {
            color: var(--grey-light);
        }
    }
}



@media (max-width: 768px) {
    .footer {
        padding: 0 2rem;

        .socialMidia {
            gap: 1rem;
        }
    }
}

@media (max-width: 414px) {

    .footer {
        justify-content: flex-start;
        padding: 0 1.2rem;

        .socialMidia {
            display: none;
        }
    }
}