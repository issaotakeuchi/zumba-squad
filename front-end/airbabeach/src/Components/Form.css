.formulario {
  margin: 0 auto;
  max-width: 29rem;
  width: 100%;
  padding: 5rem 1rem;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  gap: 0.75rem;
  -webkit-animation-duration: 0.5s;
          animation-duration: 0.5s;
  -webkit-transition-timing-function: ease-in;
          transition-timing-function: ease-in;
  -webkit-animation-fill-mode: both;
          animation-fill-mode: both;
  -webkit-animation-name: fadeIn;
          animation-name: fadeIn;
}

.formulario h1 {
  color: var(--green-light);
  margin-bottom: 0.75rem;
}

.formulario .inputs {
  max-width: 28rem;
  width: 100%;
}

.formulario .inputs label {
  display: block;
  height: 1.375rem;
}

.formulario .inputs input {
  padding: 0 0.875rem;
  border: 0;
  display: block;
  width: 100%;
  height: 2.5rem;
  background: #FFFFFF;
  border-radius: 5px;
  -webkit-box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
          box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
}

.formulario .inputs input[type="password"] {
  font-size: 20px;
}

.formulario .inputs input:focus {
  outline: transparent;
  -webkit-box-shadow: 0 0 0 2px var(--green-light);
          box-shadow: 0 0 0 2px var(--green-light);
}

.formulario .inputs .icontainer {
  max-height: 0;
  padding-right: 0.75rem;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
      -ms-flex-pack: end;
          justify-content: flex-end;
  position: relative;
  top: -2rem;
}

.formulario .inputs .icontainer .iconStyle {
  cursor: pointer;
  color: var(--green-light);
  -webkit-transform: rotateX(180deg);
          transform: rotateX(180deg);
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
}

.formulario .inputs .icontainer .iconStyle:hover {
  color: #607D8B;
}

.formulario .containerInputs {
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  gap: 0.75rem;
}

.formulario .containerInputs .errorContainer {
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  gap: 0.75rem;
}

.formulario .errorMsg {
  color: #FF0000;
  -ms-flex-item-align: end;
      align-self: flex-end;
  -webkit-animation-duration: 0.8s;
          animation-duration: 0.8s;
  -webkit-transition-timing-function: ease-in;
          transition-timing-function: ease-in;
  -webkit-animation-fill-mode: both;
          animation-fill-mode: both;
  -webkit-animation-name: fadeIn;
          animation-name: fadeIn;
}

.formulario .btn {
  margin: 3.25rem 0 1rem 0;
  max-width: 13rem;
  width: 100%;
  min-height: 2.5rem;
  height: 100%;
  padding: 0.5rem 2.25rem;
  border: none;
  border-radius: 5px;
  background-color: var(--green-light);
  -webkit-box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.12);
          box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.12);
  color: var(--gray-light);
  font-style: normal;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.25rem;
  cursor: pointer;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.formulario .btn:hover {
  background-color: var(--green-medium);
}

.formulario .linkStyle {
  color: #4285F4;
  text-decoration: none;
  margin-left: 0.25rem;
}

@media (max-width: 768px) {
  .formulario {
    max-width: 22rem;
  }
  .formulario .containerInputs {
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
  }
  .formulario .containerInputs .errorSpace {
    display: none;
  }
  .formulario .btn {
    max-width: 22rem;
  }
}

@media (max-width: 414px) {
  .formulario {
    max-width: 18.75rem;
  }
  .formulario .btn {
    max-width: 18.75rem;
  }
}
/*# sourceMappingURL=Form.css.map */