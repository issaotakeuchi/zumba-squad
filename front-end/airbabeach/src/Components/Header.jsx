import { useState } from "react";
import { Sidebar } from "./Sidebar";
import { X } from 'phosphor-react'
import "./Header.scss";

export function Header() {
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
  const [isLogged, setLogged] = useState(true);

  function handleHamburgerClick() {
    setIsSidebarOpen(!isSidebarOpen);
  }

  return (
    <header className="headerFull">
      <div className="logoHolder">
        <img src="src/img/logo.png" alt="logotipo" />
        <h2 className="logoTagline">Sinta-se em casa</h2>
      </div>

      <div className="asideHolder">

        {isLogged && (
          <div className="loggedIn">
            <div className="profilePicture">
              <img src="src/img/profile.png" alt="logotipo" />
            </div>
            <div className="greetingAndName">
              <p className="">Olá,</p>
              <p className="greetingAndNameGreen">Bruno Rocha</p>
            </div>
            <div className="btnLoggout">
              <X size={26} onClick={setLogged} alt="Sair" color="var(--grey-darkest)" weight="bold" id="btnFechar" />
            </div>
          </div>
        )}


        <div className="hamburger" onClick={handleHamburgerClick}>
        <Sidebar />
        </div>

        {!isLogged && (
          <div className="btnHolder">
            <button className="btn" type="submit">
              Criar conta
            </button>
            <button className="btn" type="submit">
              Iniciar sessão
            </button>
          </div>
        )}

      </div>
    </header>
  );
}
