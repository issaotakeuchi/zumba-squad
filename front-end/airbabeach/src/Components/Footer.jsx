import "./Footer.scss"
import { FacebookLogo, LinkedinLogo, TwitterLogo, InstagramLogo } from 'phosphor-react'

export function Footer() {
    let year = new Date().getFullYear()

    return (

        <footer className="footer">
            <h4 className="footerParagraph">&#169;{year} AirBnBeach, inc </h4>

            <div className="socialMidia">
                <a href="#"><FacebookLogo size={28}  weight="fill" /></a>
                
                <a href="#"><LinkedinLogo size={25}  weight="bold" /></a>

                <a href="#"><TwitterLogo size={24}  weight="fill" /></a>

                <a href="#"><InstagramLogo size={25}  /></a>
            </div>
        </footer>

    )
}