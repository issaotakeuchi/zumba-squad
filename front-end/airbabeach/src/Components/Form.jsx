import { useState } from "react";
import axios from "axios";
import './Form.scss'
import { toast } from 'react-toastify';


import { Link, useNavigate } from "react-router-dom";
import { Eye, EyeSlash } from 'phosphor-react'
import { useAuth } from "../contexts/auth";

export function Form({ type }) {

    const { saveToken } = useAuth();
    const navigate = useNavigate();

    const strongRegex = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#\$%\^&\*])(?=.*[0-9]).{8,}$");
    const [firstName, setFirstName] = useState("");
    const [surname, setSurname] = useState("");
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [confirnPassword, setConfirnPassword] = useState("");
    const [show, setShow] = useState(false);

    const [status, setStatus] = useState({
        type: '',
        message: ''
    })

    function cleanForm() {
        if (type === 'login') {
            setEmail('')
            setPassword('')
        } else {
            setFirstName('')
            setSurname('')
            setEmail('')
            setPassword('')
            setConfirnPassword('')
        }
    }

    function changeVisibility() {
        setShow(!show)
    }

    function handleSubmit(e) {
        e.preventDefault();
        if (!validate()) return;

        let url;
        if (type === 'login') {
            url = 'https://www.airbabeach/login'
        } else {
            url = 'https://www.airbabeach/createuser'
        }

        let data;
        if (type === 'login') {
            data = {
                email: email,
                password: password
            }
        } else {
            data = {
                firstName: firstName,
                surname: surname,
                email: email,
                password: password
            }
        }

        const options = {
            headers: { 'X-Custom-Header': 'value' }
        }

        axios.post(url, data, options).then((response) => {
            console.log(response);

            if (type === 'login') {
                saveToken(response.token)
                saveUser(response.user)
                toast.success("Usuário logado com sucesso")
                navigate('/home')
            } else {
                toast.success("Usuário criado com sucesso")
                navigate('/login')
            }

        }, (error) => {
            console.log(error.code);

            if (type === 'login') {
                //if (error.status == 404) return setStatus({ type: 'loginError', message: 'Usuário não encontrado' });
                //if (error.status == 404) return setStatus({ type: 'loginError', message: 'Usuário ou senha não encontrados.' });
                //saveUser()
                if (error.status == 404) return toast.error('Usuário não encontrado');
                if (error.status == 404) return toast.error('Usuário ou senha não encontrados');
                if (error.code === 'ERR_NETWORK') return  ;

            } else {
                //if (error.status == 404) return setStatus({ type: 'loginError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                if (error.status == 404) return toast.error('Erro ao preencher o formuário. Recarregue a página e tente novamente.');
                if (error.code === 'ERR_NETWORK') return toast.error('Verifique a sua conexão com a internet.');
            }
        });

        cleanForm();
    }

    function validate(input) {

        switch (input) {
            case 'firstName':
                if (firstName === '') return setStatus({ type: 'firstNameError', message: 'Este campo é obrigatório' });
                if (firstName.length < 4) return setStatus({ type: 'firstNameError', message: 'O nome precisa ter mais de 3 letras' });
                if (firstName == undefined || firstName === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                break;
            case 'surname':
                if (surname === '') return setStatus({ type: 'surnameError', message: 'Este campo é obrigatório' });
                if (surname.length < 4) return setStatus({ type: 'surnameError', message: 'O sobrenome precisa ter mais de 3 letras' });
                if (surname == undefined || surname === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                break;
            case 'email':
                if (email === '') return setStatus({ type: 'emailError', message: 'Este campo é obrigatório' });
                if (email == undefined || email === null) return setStatus({ type: 'emailError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                break;
            case 'password':
                if (password === '') return setStatus({ type: 'passwordError', message: 'Este campo é obrigatório' });
                if (type === 'login') {
                    if (password === '') return setStatus({ type: 'passwordError', message: 'Este campo é obrigatório' });
                    if (password == undefined || password === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                } else {
                    if (password === '') return setStatus({ type: 'passwordError', message: 'Este campo é obrigatório' });
                    if (!password.match(strongRegex)) return setStatus({ type: 'passwordError', message: 'A senha deve conter pelo menos um caracter especial, número, 7 letras, letra maiúscula e múscula ' });
                    if (confirnPassword !== '' && confirnPassword !== password) return setStatus({ type: 'confirnPasswordError', message: 'As duas senhas devem ser iguais' });
                    if (password == undefined || password === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                }
                break;
            case 'confirnPassword':
                if (confirnPassword === '') return setStatus({ type: 'confirnPasswordError', message: 'Este campo é obrigatório' });
                if (confirnPassword !== password) return setStatus({ type: 'confirnPasswordError', message: 'As duas senhas devem ser iguais' });
                if (confirnPassword == undefined || confirnPassword === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                break;
            default:
                if (type === 'login') {
                    if (email === '') return setStatus({ type: 'emailError', message: 'Este campo é obrigatório' });
                    if (email == undefined || email === null) return setStatus({ type: 'emailError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                    if (password === '') return setStatus({ type: 'passwordError', message: 'Este campo é obrigatório' });
                    if (password == undefined || password === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                } else {
                    if (firstName === '') return setStatus({ type: 'firstNameError', message: 'Este campo é obrigatório' });
                    if (firstName.length < 4) return setStatus({ type: 'firstNameError', message: 'O nome precisa ter mais de 3 letras' });
                    if (firstName == undefined || firstName === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                    if (surname === '') return setStatus({ type: 'surnameError', message: 'Este campo é obrigatório' });
                    if (surname.length < 4) return setStatus({ type: 'surnameError', message: 'O sobrenome precisa ter mais de 3 letras' });
                    if (surname == undefined || surname === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                    if (email === '') return setStatus({ type: 'emailError', message: 'Este campo é obrigatório' });
                    if (email == undefined || email === null) return setStatus({ type: 'emailError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                    if (password === '') return setStatus({ type: 'passwordError', message: 'Este campo é obrigatório' });
                    if (!password.match(strongRegex)) return setStatus({ type: 'passwordError', message: 'A senha deve conter pelo menos um caracter especial, número, 7 letras, letra maiúscula e múscula ' });
                    if (password == undefined || password === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                    if (confirnPassword === '') return setStatus({ type: 'confirnPasswordError', message: 'Este campo é obrigatório' });
                    if (confirnPassword !== password) return setStatus({ type: 'confirnPasswordError', message: 'As duas senhas devem ser iguais' });
                    if (confirnPassword == undefined || confirnPassword === null) return setStatus({ type: 'passwordError', message: 'Erro ao preencher o formuário. Recarregue a página e tente novamente.' });
                }
                break;
        }

        //apagar o status da mensagen
        setStatus({
            type: '',
            message: ''
        })
        return true;
    }

    return (
        <form className="formulario" onSubmit={handleSubmit}>
            {type && type === "createUser" && <h1>Criar conta</h1>}
            {type && type === "login" && <h1>Iniciar Sessão</h1>}


            {type && type === "createUser" &&
                <div className="containerInputs">
                    <div className="errorContainer">
                        <div className="inputs">
                            <label htmlFor="firstName" className="text-small">Nome</label>
                            <input
                                className="text-small"
                                type="text"
                                name="firstName"
                                id="firstName"
                                value={firstName}
                                onChange={(e) => setFirstName(e.target.value)}
                                onBlur={() => validate('firstName')}
                            />
                        </div>
                        {status.type === 'firstNameError' ? <p className="text-small errorMsg">{status.message}</p> : ""}
                        {status.type === 'surnameError' ? <p className="text-small errorSpace" style={{ color: '#ffffff00' }}>{status.message}</p> : ""}
                    </div>

                    <div className="errorContainer">
                        <div className="inputs">
                            <label htmlFor="surname" className="text-small">Sobrenome</label>
                            <input
                                className="text-small"
                                type="text"
                                name="surname"
                                id="surname"
                                value={surname}
                                onChange={(e) => setSurname(e.target.value)}
                                onBlur={() => validate('surname')}
                            />
                        </div>
                        {status.type === 'surnameError' ? <p className="text-small errorMsg">{status.message}</p> : ""}
                        {status.type === 'firstNameError' ? <p className="text-small errorSpace" style={{ color: '#ffffff00' }}>{status.message}</p> : ""}
                    </div>
                </div>
            }

            <div className="inputs">
                <label htmlFor="email" className="text-small">Email</label>
                <input
                    className="text-small"
                    type="email"
                    name="email"
                    id="email"
                    value={email}
                    onChange={(e) => setEmail(e.target.value)}
                    onBlur={() => validate('email')}
                />
            </div>
            {status.type === 'emailError' ? <p className="text-small errorMsg">{status.message}</p> : ""}

            <div className="inputs">
                <label htmlFor="password" className="text-small">Senha</label>
                <input
                    className="text-small"
                    type={show ? 'text' : 'password'}
                    name="password"
                    id="password"
                    value={password}
                    onChange={(e) => setPassword(e.target.value)}
                    onBlur={() => validate('password')}
                />
                <div className="icontainer">
                    {
                        show ?
                            <Eye size={24} className="iconStyle" onClick={changeVisibility} /> :
                            <EyeSlash size={24} className="iconStyle" onClick={changeVisibility} />
                    }
                </div>
            </div>
            {status.type === 'passwordError' ? <p className="text-small errorMsg">{status.message}</p> : ""}


            {type && type === "createUser" &&
                <>
                    <div className="inputs">
                        <label htmlFor="confirnPassword" className="text-small">Confirmar senha</label>
                        <input
                            className="text-small"
                            type={show ? 'text' : 'Password'}
                            name="confirnPassword"
                            id="confirnPassword"
                            value={confirnPassword}
                            onChange={(e) => setConfirnPassword(e.target.value)}
                            onBlur={() => validate('confirnPassword')}
                        />
                    </div>
                    {status.type === 'confirnPasswordError' ? <p className="text-small errorMsg" style={{ alignSelf: 'flex-end' }}>{status.message}</p> : ""}
                </>
            }

            {type && type === "createUser" &&
                <>
                    <button className="btn" type="submit"> Criar conta </button>
                    <p className="text-small">
                        Já tem uma conta?
                        <Link to={'/login'} className='linkStyle'>Iniciar sessão</Link>
                    </p>
                </>
            }

            {type && type === "login" &&
                <>
                    <button className="btn" type="submit"> Entrar </button>
                    <p className="text-small">
                        Ainda não tem conta?
                        <Link to={'/createUser'} className='linkStyle'>Registre-se.</Link>
                    </p>
                </>
            }

        </form>

    )
}