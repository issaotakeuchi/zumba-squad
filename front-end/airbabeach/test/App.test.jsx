import { describe, it, expect, beforeAll, afterAll } from 'vitest';
import { render, screen, fireEvent, waitFor, getByRole } from './test-utils.jsx';

import { Home } from '../src/pages/Home';
import { Footer } from '../src/Components/Footer'
import { Header } from '../src/Components/Header'
import { Form } from '../src/Components/Form'

describe('Home component', () => {
  it('É renderizado corretamente"', () => {
    render(<Home />);
    //screen.debug();
    expect(screen.getByText('Home')).toBeInTheDocument();
  });
});


describe('Header component', () => {
  it('É renderizado corretamente"', () => {
    render(<Header />);
    //screen.debug();
    expect(screen.getByText('header')).toBeInTheDocument();
    //expect(screen.getByText(/©2023 AirBnBeach, inc/)).toMatchSnapshot()
    //expect(screen.getByText(/Content/i)).toBeUndefined()
  });
});


describe('Footer component', () => {
  it('É renderizado corretamente"', () => {
    render(<Footer />);
    expect(screen.getByText(/©2023 AirBnBeach, inc/)).toBeInTheDocument();

  });
});

describe('Login-Form component', () => {
  it('É renderizado corretamente"', () => {
    render(<Form type='login' />);
    //screen.debug();
    expect(screen.getByText(/Iniciar Sessão/)).toBeInTheDocument();

  });
});

describe('CreateUser-Form component', () => {
  it('É renderizado corretamente', () => {
    render(<Form type='createUser' />);
    //Tratativa para lidar com múltiplos elementos quando se quer um em específico 
    expect(screen.getAllByText(/Criar conta/)[0]).toBeInTheDocument();
  }
  );



  /*  it("Clicar no botão login é direcionado a página de login", async () => {
     
     render(<Form type='createUser'/>);
     
 
     fireEvent.click(screen.getByTestId('Homac'))
     //fireEvent.click(screen.getByText('Soma'))
 
     
     expect(screen.debug())
 
 
     
       
     //screen.debug()
     //expect(screen.getByText('Home')).toBeInTheDocument();
 
     
 }); */



});








