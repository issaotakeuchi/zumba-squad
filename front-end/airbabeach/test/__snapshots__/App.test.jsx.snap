// Vitest Snapshot v1

exports[`Footer > É renderizado corretamente" 1`] = `
<h4
  class="footerParagraph"
>
  ©
  2023
   AirBnBeach, inc 
</h4>
`;
